src/cts/src/Clock.h

```diff
@@ -218,6 +223,44 @@ class Clock
     return subNets_.back();
   }
 
+  void removeSubNet(ClockSubNet sub_net)
+  {
+    std::set<ClockInst*> sinksToRemove;
+    sinksToRemove.insert(sub_net.getDriver());
+    for (auto it = subNets_.begin(); it != subNets_.end();) {
+      if (it->getName() == sub_net.getName()) {
+        auto parent_net = it->getParentSubNet();
+        it = subNets_.erase(it);
+        if (parent_net == nullptr) {
+          break;
+        }
+        //    If the parent subnet is dangling after removing the driving
+        //    buffer,
+        //  also remove the parent subnet
+        parent_net->removeSinks(sinksToRemove);
+        if (parent_net->getNumSinks() == 0) {
+          removeSubNet(*parent_net);
+        }
+        break;
+      } else {
+        ++it;
+      }
+    }
+    removeClockBuffer(sub_net.getDriver());
+  }
+
+  void removeClockBuffer(ClockInst* clock_buffer)
+  {
+    for (auto it = clockBuffers_.begin(); it != clockBuffers_.end();) {
```

Comment:
Similar question here. Isn't there a way to avoid these loops? Perhaps a hash table, a map...